# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1455130789 18000
#      Wed Feb 10 13:59:49 2016 -0500
# Node ID c2b17840910e8323f24bbdb5bf3847474d4ff201
# Parent  ae03a16f334380070ecd03155767a3458b76b538
generate and install smx-ops.h and vx-ops.h

* liboctave/module.mk (BUILT_INCS):
Include smx-ops.h and vx-ops.h in the list.
* liboctave/operators/module.mk (run-mx-ops-inclusive): New macro.
(liboctave/operators/smx-ops.h, liboctave/operators/vx-ops.h):
New targets.
(liboctave/operators/mx-ops.h, liboctave/operators/smx-ops.h,
liboctave/operators/vx-ops.h): Use run-mx-ops-inclusive.

diff --git a/liboctave/module.mk b/liboctave/module.mk
--- a/liboctave/module.mk
+++ b/liboctave/module.mk
@@ -19,16 +19,18 @@ liboctave_liboctave_la_CPPFLAGS = \
 liboctave_liboctave_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
 
 liboctave_liboctave_la_CXXFLAGS = $(AM_CXXFLAGS) $(WARN_CXXFLAGS)
 
 octlib_LTLIBRARIES += liboctave/liboctave.la
 
 BUILT_INCS = \
   liboctave/operators/mx-ops.h \
+  liboctave/operators/smx-ops.h \
+  liboctave/operators/vx-ops.h \
   $(LIBOCTAVE_OPT_INC) \
   $(MX_OP_INC) \
   $(VX_OP_INC) \
   $(SMX_OP_INC)
 
 BUILT_SOURCES += $(BUILT_INCS)
 
 octinclude_HEADERS += \
diff --git a/liboctave/operators/module.mk b/liboctave/operators/module.mk
--- a/liboctave/operators/module.mk
+++ b/liboctave/operators/module.mk
@@ -41,20 +41,30 @@ endef
 	$(AM_V_GEN)$(call run-mx-ops,vx)
 
 $(MX_OP_INC) $(MX_OP_SRC) : liboctave/operators/mk-ops.awk liboctave/operators/mx-ops
 	$(AM_V_GEN)$(call run-mx-ops,mx)
 
 $(SMX_OP_INC) $(SMX_OP_SRC) : liboctave/operators/sparse-mk-ops.awk liboctave/operators/smx-ops
 	$(AM_V_GEN)$(call run-mx-ops,smx,sparse-)
 
+define run-mx-ops-inclusive
+  rm -f $@-t $@ && \
+  $(AWK) -f $(OP_SRCDIR)/$(notdir $<) prefix=$(patsubst %-ops.h,%,$(notdir $@)) make_inclusive_header=$(notdir $@) $(OP_SRCDIR)/$(basename $(notdir $@)) > $@-t && \
+	mv $@-t $@
+endef
+
+liboctave/operators/vx-ops.h : liboctave/operators/mk-ops.awk liboctave/operators/vx-ops
+	$(AM_V_GEN)$(run-mx-ops-inclusive)
+
 liboctave/operators/mx-ops.h : liboctave/operators/mk-ops.awk liboctave/operators/mx-ops
-	$(AM_V_GEN)rm -f $@-t $@ && \
-	$(AWK) -f $(OP_SRCDIR)/mk-ops.awk prefix=mx make_inclusive_header=mx-ops.h $(OP_SRCDIR)/mx-ops > $@-t && \
-	mv $@-t $@
+	$(AM_V_GEN)$(run-mx-ops-inclusive)
+
+liboctave/operators/smx-ops.h : liboctave/operators/sparse-mk-ops.awk liboctave/operators/smx-ops
+	$(AM_V_GEN)$(run-mx-ops-inclusive)
 
 noinst_LTLIBRARIES += liboctave/operators/liboperators.la
 
 liboctave_operators_liboperators_la_SOURCES = $(LIBOCTAVE_OPERATORS_SRC)
 
 nodist_liboctave_operators_liboperators_la_SOURCES = $(BUILT_LIBOCTAVE_OPERATORS_SOURCES)
 
 liboctave_operators_liboperators_la_CPPFLAGS = $(liboctave_liboctave_la_CPPFLAGS)
